### PR TITLE
Remove FXIOS-7979 [v122] Redux feature flag

### PR DESCRIPTION
**Pull request by yoanarios**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7979)

## :bulb: Description
- Remove Redux feature flag
- Remove unnecessary Dispatch async when dispatching actions from middleware on `ThemeManagerMiddleware` example
- Fix swiftlint warning

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Review by adudenamedruby — APPROVED** [review]
LGTM

File: Client/Frontend/Settings/ThemeSettings/ThemeMiddleware.swift [hunk 1]
```diff
@@ -27,35 +27,23 @@ class ThemeManagerMiddleware: ThemeManagerProvider {
         switch action {
         case ThemeSettingsAction.themeSettingsDidAppear:
             let currentThemeState = self.getCurrentThemeManagerState()
-            DispatchQueue.main.async {
```
Comment (lmarceau) [inline Client/Frontend/Settings/ThemeSettings/ThemeMiddleware.swift:30]: The `DispatchQueue.main.async {` aren't needed anymore?

**Review by lmarceau — APPROVED** [review]

File: Client/Frontend/Settings/ThemeSettings/ThemeMiddleware.swift (hunk 1 above)
Comment (yoanarios) [reply Client/Frontend/Settings/ThemeSettings/ThemeMiddleware.swift:30]: No, a check was added at the Store level to send the actions from middlewares using `DispatchQueue.main.async ` already so this cleanup was needed

**Comment by adudenamedruby** [comment]
@mergifyio backport release/v122

**Pull request merged**